### PR TITLE
[json-rpc] add more tests

### PR DESCRIPTION
**Pull request by xli**
1. added more corner cases test cases
2. added tests for all different types event data views
3. removed a wrapper when parsing event, the original version returns "unknown" event if parsing failed, which hides real error, changed to return server internal error to expose problem.

Still missing EventDataView::UpgradeEvent test, don't know how to generate it, will need talk to @runtian-zhou about it and add test later.

New event CreateAccountEvent is added, it is created recently (https://github.com/libra/libra/pull/6156).
Added new test to ensure there is no unknown events in integration tests at the end, to make test pass, created blank CreateAccountEvent for now (tried to decoded it, but got an error, so try to keep this diff simpler, will talk to @sblackshear  to add created account address field if necessary)

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/types/src/views.rs [hunk 1]
```diff
@@ -188,174 +191,148 @@ pub enum EventDataView {
...
+
+impl TryFrom<(u64, ContractEvent)> for EventView {
+    type Error = Error;
+
+    fn try_from((txn_version, event): (u64, ContractEvent)) -> Result<Self> {
+        Ok(EventView {
             key: BytesView::from(event.key().as_bytes()),
             sequence_number: event.sequence_number(),
             transaction_version: txn_version,
-            data: event_data.unwrap_or(EventDataView::Unknown {}),
```
Comment (matbd) [inline json-rpc/types/src/views.rs:357]: indeed

**Review by matbd — APPROVED** [review]
LGTM
